domain/tutoringModel/tyche.py

```diff
@@ -0,0 +1,303 @@
+            elif curr_le == "manuskript_ek":
+                curr_le = "EK"
+            elif curr_le == "manuskript_ab":
+                curr_le = "AB"
+            elif curr_le == "manuskript_be":
+                curr_le = "BE"
+            elif curr_le == "quiz_rq":
+                curr_le = "RQ"
+            elif curr_le == "quiz_se":
+                curr_le = "SE"
+            elif curr_le == "uebung":
+                curr_le = "ÜB"
+            elif curr_le == "zusammenfassung":
+                curr_le = "ZF"
+            elif curr_le == "zusatzmaterial_textuell":
+                curr_le = "ZL"
+            elif curr_le == "animation":
+                curr_le = "AN"
+            else:
+                print("Current learning element", curr_le, "is not part of our set!")
+            self.learningpath[i] = curr_le
+        pass
+
+    def set_last_learning_element(self, classification_learning_element):
+        if self.last_le:
+            curr_le = self.last_le
+            if curr_le == "KÜ":
+                classification_learning_element[0] = "kurzuebersicht"
+            elif curr_le == "LZ":
+                classification_learning_element[1] = "lernziele"
+            elif curr_le == "EK":
+                classification_learning_element[2] = "manuskript_ek"
+            elif curr_le == "AB":
+                classification_learning_element[3] = "manuskript_ab"
+            elif curr_le == "BE":
+                classification_learning_element[4] = "manuskript_be"
+            elif curr_le == "RQ":
+                classification_learning_element[5] = "quiz_rq"
+            elif curr_le == "SE":
+                classification_learning_element[6] = "quiz_se"
+            elif curr_le == "ÜB":
+                classification_learning_element[7] = "uebung"
+            elif curr_le == "ZF":
+                classification_learning_element[8] = "zusammenfassung"
+            elif curr_le == "ZL":
+                classification_learning_element[9] = "zusatzmaterial_textuell"
+            elif curr_le == "FO":
+                classification_learning_element[10] = "kollaborativ"
+            elif curr_le == "AN":
+                classification_learning_element[11] = "animation"
+            else:
+                print("LE", curr_le, "is not part of our set!!")
+        return classification_learning_element
+
+    def get_learning_style(self):
+        lstyle = [0, 0, 0, 0]
+
+        # Active-Reflective Dimension:
+
+        dim1 = self.learning_style["processing_dimension"]
+        if dim1 == "act":
+            lstyle[0] = "active"
+        elif dim1 == "ref":
+            lstyle[0] = "reflective"
+        else:
+            print("Invalid value!")
+        # Visual-Verbal Dimension:
+
+        dim2 = self.learning_style["input_dimension"]
+        if dim2 == "vis":
+            lstyle[1] = "visual"
+        elif dim2 == "vrb":
+            lstyle[1] = "verbal"
+        else:
+            print("Invalid value!")

```

Comment:
exception needed or just warning?

Reply:
warning is enough, if a learning style dimension is wrong, it is checked in get_learning_path()